src/scenic/core/pruning.py

```diff
@@ -565,41 +630,30 @@ def percentagePruned(base, newBase):
     return None
 
 
-def checkCyclical(A, B):
-    """Check for a potential circular dependency
-
-    Returns True if the scenario would have a circular dependency
-    if A depended on B.
-    """
-    state = collections.defaultdict(lambda: 0)
+def checkConditionedCycle(A, B):
+    """Returns true if A depends on B"""
```

Comment:
It looks to me like this won't work if (after conditioning) `A` is `B` or directly depends on it (the latter since you don't check if `target_dep` itself is `B`, only its dependencies). Presumably these cases don't arise with the current pruning methods, but we might as well make this function satisfy the description in the docstring in case we use it differently later.

Reply:
Good catch, I made a change that should fix it.